src/omnicore/rpctxobject.cpp

```diff
@@ -256,6 +256,7 @@ void populateRPCTypeTradeOffer(CMPTransaction& omniObj, Object& txobj)
 
     // Check levelDB to see if the amount for sale has been amended due to a partial purchase
     // TODO: DEx phase 1 really needs an overhaul to work like MetaDEx with original amounts for sale and amounts remaining etc
+    // TODO: if the amount of the transaction is not used here, then the BTC amount should be recalculated (?)
```

Comment:
> The (aweful) way how it's currently handled

Ugh, agreed - the source of many a consensus mismatch in the early days when we had multiple implementations :(

>  but the BTC amount isn't updated

I see, thanks for the explanation - good catch.  Perhaps it would be better to handle it just like MetaDEx offers, where the "original amounts" can be provided along with the available (modified) values:

```
{
  ...
  "amountforsale" : 1000,
  "amountremaining" : 50,
  "amountdesired" : 1000.
  "amounttofill" : 50,
  "unitprice" : "1.00000000"
  ...
}
```